src/Common/src/Interop/Unix/libssl/Interop.libssl_types.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class libssl
+    {
+        internal const int SSL_CTRL_OPTIONS = 32;
+        internal static bool OPENSSL_NO_COMP = true; //no compression true by default
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct SSL_CIPHER
+        {
+            internal int valid;
+            internal string name; /* text name */
+            internal long id; /* id, 4 bytes, first is version */
```

Comment:
All of these `long`s should really be `ulongs`; except that they really should be `UIntPtr` to get the marshal size correct.  But since we're going to move these to shims anyways it probably doesn't matter.

(FYI, @nguerrera and @sokket: a native `unsigned long` struct is being added)